src/agents/base.ts

```diff
@@ -63,6 +65,12 @@ export abstract class BaseAgent<
         }
       },
     );
+
+    if (INSTRUMENTATION_ENABLED) {
+      return run.middleware(createTelemetryMiddleware());
+    } else {
+      return run;
+    }
```

Comment:
This can be simplified to just

```
return RunContext
  .enter(...)
  .middleware(INSTRUMENTATION_ENABLED ? createTelemetryMiddleware : noop)
```